packages-exp/auth-exp/src/platform_browser/persistence/browser.ts

```diff
@@ -73,21 +81,198 @@ function remove(this: BrowserPersistenceClass, key: string): Promise<void> {
 class BrowserLocalPersistence implements BrowserPersistenceClass {
   static type: 'LOCAL' = 'LOCAL';
   type = PersistenceType.LOCAL;
+  storage = localStorage;
   isAvailable = isAvailable;
+
   set = set;
   get = get;
   remove = remove;
-  storage = localStorage;
+
+  // The polling period in case events are not supported
+  static POLLING_TIMER_INTERVAL = 1000;
+  // The IE 10 localStorage cross tab synchronization delay in milliseconds
+  static IE10_LOCAL_STORAGE_SYNC_DELAY = 10;
+
+  private readonly listeners: Record<string, StorageEventListener[]> = {};
+  private readonly localCache: Record<string, string | null> = {};
+  private pollTimer: NodeJS.Timeout | null = null;
+  
+  // Safari or iOS browser and embedded in an iframe.
+  private readonly safariLocalStorageNotSynced = _iframeCannotSyncWebStorage() && _isIframe();
+
+  _forAllChangedKeys(cb: (key: string, oldValue: string | null, newValue: string | null) => void): void {
+    // Check all keys with listeners on them.
+    Object.keys(this.listeners).forEach(key => {
```

Comment:
Don't use `Array.prototype.forEach`, use a for-of loop: go/tsstyle#iterating-containers

```ts
for (const key of Object.keys(this.listeners)) {
}
```

Reply:
done